poc_mobile_verifier/src/shares.rs

```diff
@@ -0,0 +1,127 @@
+//! Heartbeat storage
+
+use crate::Result;
+use chrono::{DateTime, Utc};
+use file_store::{file_sink, FileStore};
+use file_store::heartbeat::CellHeartbeat;
+use helium_proto::services::poc_mobile as proto;
+use std::ops::Range;
+
+use crate::cell_type::CellType;
+
+pub struct Shares {
+    pub invalid_shares: Vec<proto::Share>,
+    pub valid_shares: Vec<proto::Share>,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum ValidateHeartbeatsError {
+    #[error("file store error: {0}")]
+    FileStoreError(#[from] file_store::Error),
+    #[error("database error: {0}")]
+    DbError(#[from] sqlx::Error),
+}
+
+impl Shares {
+    fn new() -> Self {
+        Self {
+            valid_shares: Vec::new(),
+            invalid_shares: Vec::new(),
+        }
+    }
+
+    pub async fn validate_heartbeats(
+        file_store: &FileStore,
+        epoch: &Range<DateTime<Utc>>,
+    ) -> Result<Self> {
+
+        let shares = crate::ingest::new_heartbeat_reports(file_store, epoch)
+            .await?
+            .iter()
+            .map(|ingest_heartbeat| to_share(ingest_heartbeat, epoch))
+            .fold(Shares::new(), |mut acc, share| {
+                if share.validity == proto::ShareValidity::Valid as i32 {
+                    acc.valid_shares.push(share);
+                } else {
+                    acc.invalid_shares.push(share);
+                }
+                acc
+            });
+
+        Ok(shares)
+    }
+
+    pub async fn write(
+        self,
+        valid_shares_tx: &file_sink::MessageSender,
+        invalid_shares_tx: &file_sink::MessageSender,
+    ) -> Result<()> {
```

Comment:
is this redundant when you've imported the crate-local `Result` type?

Reply:
yup